### DISCUSSION TITLE
Attachments not working anymore

### DISCUSSION THREAD

**Discussion by HellStorm666** [dani-garcia/vaultwarden#4967] [Q&A]
Hi all,

If I try to upload an attachment, via the webbrowser, plugin or windows client, Vaultwarden crashes and I get logged out.
Also downloading an attachment is not working all the time. (some attachments, like my SSH .pub key are not working. most .pdf's are working)

What is wrong?

Only think that has changed is that Traefik has updated from V2 to V3.
But all other traefik routed stuff still works.

Diagnostics says all is ok.

Below the support string:

> ### Your environment (Generated via diagnostics page)
> * Vaultwarden version: v1.32.0
> * Web-vault version: v2024.6.2b
> * OS/Arch: linux/x86_64
> * Running within a container: true (Base: Debian)
> * Environment settings overridden: true
> * Uses a reverse proxy: true
> * IP Header check: true (X-Forwarded-For)
> * Internet access: true
> * Internet access via a proxy: false
> * DNS Check: true
> * Browser/Server Time Check: true
> * Server/NTP Time Check: true
> * Domain Configuration Check: true
> * HTTPS Check: true
> * Database type: SQLite
> * Database version: 3.46.0
> * Clients used: 
> * Reverse proxy and version: 
> * Other relevant information: 
> 
> ### Config (Generated via diagnostics page)
> <details><summary>Show Running Config</summary>
> 
> **Environment settings which are overridden:** DOMAIN, ADMIN_TOKEN
> 
> 
> ```json
> {
>   "_duo_akey": null,
>   "_enable_duo": false,
>   "_enable_email_2fa": true,
>   "_enable_smtp": true,
>   "_enable_yubico": true,
>   "_icon_service_csp": "",
>   "_icon_service_url": "",
>   "_ip_header_enabled": true,
>   "_max_note_size": 10000,
>   "_smtp_img_src": "cid:",
>   "admin_ratelimit_max_burst": 3,
>   "admin_ratelimit_seconds": 300,
>   "admin_session_lifetime": 20,
>   "admin_token": "***",
>   "allowed_iframe_ancestors": "",
>   "attachments_folder": "data/attachments",
>   "auth_request_purge_schedule": "30 * * * * *",
>   "authenticator_disable_time_drift": false,
>   "data_folder": "data",
>   "database_conn_init": "",
>   "database_max_conns": 10,
>   "database_timeout": 30,
>   "database_url": "***************",
>   "db_connection_retries": 15,
>   "disable_2fa_remember": false,
>   "disable_admin_token": false,
>   "disable_icon_download": false,
>   "domain": "*****://************************",
>   "domain_origin": "*****://************************",
>   "domain_path": "",
>   "domain_set": true,
>   "duo_context_purge_schedule": "30 * * * * *",
>   "duo_host": null,
>   "duo_ikey": null,
>   "duo_skey": null,
>   "duo_use_iframe": false,
>   "email_2fa_auto_fallback": false,
>   "email_2fa_enforce_on_verified_invite": false,
>   "email_attempts_limit": 3,
>   "email_change_allowed": true,
>   "email_expiration_time": 600,
>   "email_token_size": 6,
>   "emergency_access_allowed": true,
>   "emergency_notification_reminder_schedule": "0 3 * * * *",
>   "emergency_request_timeout_schedule": "0 7 * * * *",
>   "enable_db_wal": true,
>   "enable_websocket": true,
>   "event_cleanup_schedule": "0 10 0 * * *",
>   "events_days_retain": null,
>   "experimental_client_feature_flags": "fido2-vault-credentials",
>   "extended_logging": true,
>   "helo_name": null,
>   "hibp_api_key": "***",
>   "http_request_block_non_global_ips": true,
>   "http_request_block_regex": null,
>   "icon_blacklist_non_global_ips": true,
>   "icon_blacklist_regex": null,
>   "icon_cache_folder": "data/icon_cache",
>   "icon_cache_negttl": 259200,
>   "icon_cache_ttl": 2592000,
>   "icon_download_timeout": 10,
>   "icon_redirect_code": 302,
>   "icon_service": "internal",
>   "incomplete_2fa_schedule": "30 * * * * *",
>   "incomplete_2fa_time_limit": 3,
>   "increase_note_size_limit": false,
>   "invitation_expiration_hours": 120,
>   "invitation_org_name": "REDACTED",
>   "invitations_allowed": true,
>   "ip_header": "X-Forwarded-For",
>   "job_poll_interval_ms": 30000,
>   "log_file": null,
>   "log_level": "info",
>   "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
>   "login_ratelimit_max_burst": 10,
>   "login_ratelimit_seconds": 60,
>   "org_attachment_limit": null,
>   "org_creation_users": "",
>   "org_events_enabled": false,
>   "org_groups_enabled": false,
>   "password_hints_allowed": true,
>   "password_iterations": 1000000,
>   "push_enabled": true,
>   "push_identity_uri": "https://identity.bitwarden.eu",
>   "push_installation_id": "***",
>   "push_installation_key": "***",
>   "push_relay_uri": "https://api.bitwarden.eu",
>   "reload_templates": false,
>   "require_device_email": false,
>   "rsa_key_filename": "data/rsa_key",
>   "send_purge_schedule": "0 5 * * * *",
>   "sendmail_command": null,
>   "sends_allowed": true,
>   "sends_folder": "data/sends",
>   "show_password_hint": true,
>   "signups_allowed": false,
>   "signups_domains_whitelist": "",
>   "signups_verify": true,
>   "signups_verify_resend_limit": 6,
>   "signups_verify_resend_time": 3600,
>   "smtp_accept_invalid_certs": false,
>   "smtp_accept_invalid_hostnames": false,
>   "smtp_auth_mechanism": "Login",
>   "smtp_debug": false,
>   "smtp_embed_images": true,
>   "smtp_explicit_tls": null,
>   "smtp_from": "**********************",
>   "smtp_from_name": "Bitwarden_RS",
>   "smtp_host": "******************",
>   "smtp_password": "***",
>   "smtp_port": 587,
>   "smtp_security": "starttls",
>   "smtp_ssl": null,
>   "smtp_timeout": 15,
>   "smtp_username": "**********************",
>   "templates_folder": "data/templates",
>   "tmp_folder": "data/tmp",
>   "trash_auto_delete_days": null,
>   "trash_purge_schedule": "0 5 0 * * *",
>   "use_sendmail": false,
>   "use_syslog": false,
>   "user_attachment_limit": null,
>   "user_send_limit": null,
>   "web_vault_enabled": true,
>   "web_vault_folder": "web-vault/",
>   "yubico_client_id": "50797",
>   "yubico_secret_key": "***",
>   "yubico_server": null
> }
> ```
> </details>

And our compose config (stack in portainer)

> version: '3.8'
> 
> services:
>   vaultwarden:
>     image: "vaultwarden/server:latest"
>     container_name: "vaultwarden"
>     hostname: "vaultwarden"
>     restart: always
>     user: "1001:1000"
>     networks:
>       - APPS
>     environment:
>       - DOMAIN=https://REDACTED
>       - ADMIN_TOKEN=REDACTED
>       - ROCKET_PROFILE=release
>       - ROCKET_ADDRESS=0.0.0.0
>       - ROCKET_PORT=80
>       - ROCKET_WORKERS=20
>       - PUSH_ENABLED=true
>       - PUSH_INSTALLATION_ID=REDACTED
>       - PUSH_INSTALLATION_KEY=REDACTED
>       #- PUSH_RELAY_BASE_URI=https://push.bitwarden.com
>       - PUSH_RELAY_URI=https://api.bitwarden.eu
>       - PUSH_IDENTITY_URI=https://identity.bitwarden.eu
>     volumes:
>       - /home/REDACTED/docker/vaultwarden:/data
>     labels:
>       - "traefik.enable=true"
>       - "traefik.docker.network=APPS"
>       - "traefik.http.middlewares.redirect-https.redirectScheme.scheme=https"
>       - "traefik.http.middlewares.redirect-https.redirectScheme.permanent=true"
>       - "traefik.http.routers.bitwarden-ui-https.rule=Host(`REDACTED`)"
>       - "traefik.http.routers.bitwarden-ui-https.entrypoints=443"
>       - "traefik.http.routers.bitwarden-ui-https.tls=true"
>       - "traefik.http.routers.bitwarden-ui-https.service=bitwarden-ui"
>       - "traefik.http.services.bitwarden-ui.loadbalancer.server.port=80"
> 
> networks:
>   APPS:
>     external: true

**Comment by gdi0110** [comment]
I had a similar problem; I could upload attachments but could not download them. Going to the Admin >> Diagnostics page helped me resolve my issue. Mine was that I'm using a non-standard port and did not include the Port # in the `DOMAIN` environment variable. Once I added the Port # to the variable, I was able to download attachments.

This is what the Diagnostics page showed before:
<img width="981" alt="Screenshot 2024-09-21 at 5 55 28 PM" src="https://github.com/user-attachments/assets/116505fc-7650-4621-a78e-cf27ad89b0cf">

After changing the `DOMAIN` environment variable:
<img width="978" alt="Screenshot 2024-09-21 at 5 56 06 PM" src="https://github.com/user-attachments/assets/4be1c1f8-9b39-4fd6-ae9d-a1d05cbf3b4d">
